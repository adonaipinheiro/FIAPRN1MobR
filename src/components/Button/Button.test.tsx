import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import renderer from 'react-test-renderer';

import {Button} from './';

const mockFn = jest.fn();

const props = {
  text: 'Entrar',
};

const propsLoading = {
  text: 'Entrar',
  loading: true,
};

const propsNotPrimary = {
  text: 'Entrar',
  primary: false,
  onPress: mockFn,
};

describe('<Button />', () => {
  it('Should render corretly', () => {
    const button = renderer.create(<Button {...props} />);
    expect(button).toBeTruthy();
  });

  it('Should render corretly with loading', () => {
    const button = renderer.create(<Button {...propsLoading} />);
    expect(button).toBeTruthy();
  });

  it('Should render corretly with primary false', () => {
    const button = renderer.create(<Button {...propsNotPrimary} />);
    expect(button).toBeTruthy();
  });

  it('Should render corretly with text', () => {
    const inst = renderer.create(<Button {...propsNotPrimary} />);
    const text = inst.root.findByType(Text);
    expect(text.props.children).toBe(propsNotPrimary.text);
  });

  it('Should render corretly with onPress()', () => {
    const inst = renderer.create(<Button {...propsNotPrimary} />);
    const button = inst.root.findByType(TouchableOpacity);
    button.props.onPress();
    expect(button.props.onPress).toBeCalledTimes(1);
    expect(button).toBeTruthy();
  });

  it('Should take Snapshot', () => {
    const snapshot = renderer.create(<Button {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
