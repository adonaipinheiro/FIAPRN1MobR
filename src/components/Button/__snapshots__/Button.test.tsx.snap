// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should take Snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f2f2f2",
      "borderRadius": 4,
      "height": 40,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 10,
      "width": "100%",
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 14,
          "fontWeight": "bold",
        },
        {
          "color": "#ba1247",
        },
      ]
    }
  >
    Entrar
  </Text>
</View>
`;
